pkg/manifests/config.go

```diff
@@ -358,19 +369,27 @@ func UnmarshalStrict(data []byte, v interface{}) error {
 	}
 }
 
-func newConfig(content []byte, collectionProfilesFeatureGateEnabled bool) (*Config, error) {
+func newConfig(content []byte, collectionProfilesFeatureGateEnabled bool) (*Config, *InvalidConfigWarning, error) {
 	c := Config{CollectionProfilesFeatureGateEnabled: collectionProfilesFeatureGateEnabled}
+
+	var warning *InvalidConfigWarning
+	wCmc := defaultClusterMonitoringConfiguration()
+	wErr := UnmarshalStrict(content, &wCmc)
```

Comment:
no need to propagate warnings for this.
we already capture the error below in `err := UnmarshalStrict(content, &cmc)`

Reply:
now the newer/stricter one is the default one and we can (should) propagate its failures as errors.